tortoisebeacon/weakcoin/weak_coin.go

```diff
@@ -59,27 +58,42 @@ type weakCoin struct {
 }
 
 // a function to verify the message with the signature and its public key.
-type verifierFunc = func(msg, sig, pub []byte) (bool, error)
+type verifierFunc = func(pub, msg, sig []byte) bool
 
 type signer interface {
-	Sign(msg []byte) ([]byte, error)
+	Sign(msg []byte) []byte
+}
+
+// DefaultThreshold defines default weak coin threshold.
+// It's equal to (2^256) / 2
+// (hex: 0x8000000000000000000000000000000000000000000000000000000000000000).
+func DefaultThreshold() *big.Int {
```

Comment:
maybe we can just parse it from the hex since it is a constant?

Reply:
It was implemented as `"0x80" + strings.Repeat("00", 31))`, but I thought it might be confusing. Also, it's hard to say how many zeroes are in `0x8000000000000000000000000000000000000000000000000000000000000000` when looking at it. So I decided that `2^256 / 2` looks best. However, feel free to change it if you don't agree

Reply:
i think it is a common practice to have a copyable constant for these things. like if you have a several implementations in different languages. it is probably can be also defined in genesis config.